snooty/parser.py

```diff
@@ -854,6 +862,180 @@ def handle_method_option(self, node: n.Directive) -> None:
         if target_idx >= 0:
             node.children.insert(0, node.children.pop(target_idx))
 
+    def handle_composable(self, node: n.ComposableDirective) -> None:
+        """Handles composable directive(s) and its children composable content. Translates string options to lists for consumption"""
+
+        # first convert specified options from str -> List[str]
+        option_ids_as_string = (
+            node.options["options"] if "options" in node.options else ""
+        )
+        default_ids_as_string = (
+            node.options["defaults"] if "defaults" in node.options else ""
+        )
+        option_ids: List[str] = re.split(r"\s*,\s*", option_ids_as_string.strip())
+        default_ids: List[str] = re.split(r"\s*,\s*", default_ids_as_string.strip())
+
+        # expect at least 1 option_ids
+        if len(option_ids) < 1:
+            self.diagnostics.append(
+                InvalidChildCount(
+                    "composable-tutorial", "option_ids", "at least one", node.start[0]
+                )
+            )
+
+        # get the expected composable options from the spec
+        spec_composables = specparser.Spec.get().composables
+        spec_composables_dict = {
+            expected_option.id: expected_option for expected_option in spec_composables
+        }
+
+        # validate the specified :options: and :defaults:
+        used_ids: Set[str] = set()
+        composable_options = []
+        default_ids_dict: Dict[str, str] = {}
+        for index in range(len(option_ids)):
+            option_id = option_ids[index]
+            try:
+                self.check_valid_option_id(
+                    option_id, node, spec_composables_dict, used_ids
+                )
+                composable_from_spec = next(
+                    (option for option in spec_composables if option.id == option_id),
+                )
```

Comment:
Maybe we can use a default fallback `None` for this (next intakes a second arg as fallback value) and then have a check right after to send a specific Diagnostic?